### PR TITLE
Navitemtypechange

### LINKED ISSUE #17: Changing page type from "redirect" to "page" doesn't create empty page version
### What steps will reproduce the problem?
Change a "redirect" page to "page" type

### What is the expected result?
There should be a first empty version of the page (like when you create a new page with type "page")

### What do you get instead? (A Screenshot can help us a lot!)
An empty page without any versions in it.

### PR DESCRIPTION
**Pull request by martinpetrasch**
this fixes #17

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -13,6 +13,7 @@ All notable changes to this project will be documented in this file. This projec
 ### Fixed
 
 + [#25](https://github.com/luyadev/luya-module-cms/issues/25) Hide deleted pages in dashboard object "last page updates".
++ [#17](https://github.com/luyadev/luya-module-cms/issues/17) Create new page version if changing from page type to "page" and no version is available.
```
Comment (TheMaaarc) [inline CHANGELOG.md]: > Create new page version if changing from **other** page types to "page" and no version is available.

Teeny tiny improvement of understandability.

File: src/admin/apis/NavItemController.php [hunk 2]
```diff
@@ -353,11 +368,19 @@ public function actionUpdatePageItem($navItemId, $navItemType)
...
                     break;
             }
         }
-        
+
+        if ($model->update()) {
+            return [
+                'item' => $model,
+                'typeData' => ($model->nav_item_type == 1) ? NavItemPage::getVersionList($model->id) : $model->getType()->toArray()
+                ];
```
Comment (TheMaaarc) [inline src/admin/apis/NavItemController.php]: On tab too much.

File: src/admin/apis/NavItemPageController.php [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+ */
+class NavItemPageController extends RestActiveController
+{
+    public $modelClass = 'luya\cms\models\NavItemPage';
+
+    /**
+     * Returns all versions/pages for a given navItemId
+     *
+     * @param $navItemId
+     * @return array|\yii\db\ActiveRecord[]
```
Comment (nadar) [inline src/admin/apis/NavItemPageController.php]: use @ return array

File: src/admin/apis/NavItemController.php [hunk 4]
```diff
@@ -353,11 +368,19 @@ public function actionUpdatePageItem($navItemId, $navItemType)
                     }
                     $typeModel->insert();
                     $model->nav_item_type_id = $typeModel->id;
-                    return $model->update();
+                    break;
+                default:
+                    throw new \luya\cms\Exception("Invalid nav item type.");
```
Comment (nadar) [inline src/admin/apis/NavItemController.php]: use luya\cms\Exception in head of the file and just throw new Exception.

File: src/admin/apis/NavItemController.php [hunk 5]
```diff
@@ -325,15 +325,31 @@ public function actionUpdatePageItem($navItemId, $navItemType)
...
+                    if (!NavItemPage::find()->where(['nav_item_id' => $navItemId])->exists()) {
+                        $pageModel = new NavItemPage();
+                        $pageModel->attributes = [
+                            'nav_item_id' => $navItemId,
+                            'timestamp_create' => time(),
+                            'create_user_id' => Yii::$app->adminuser->getId(),
+                            'version_alias' => Module::t('Initial'),
+                            'layout_id' => Yii::$app->request->post('layout_id'),
+                        ];
+                        if ($pageModel->validate()) {
```
Comment (nadar) [inline src/admin/apis/NavItemController.php]:
```php
if (!$pageModel->save()) // which implied validate()
{
  return $this->sendMOdelError($pageModel);
}
```

File: src/admin/apis/NavItemController.php [hunk 6]
```diff
@@ -324,16 +324,34 @@ public function actionUpdatePageItem($navItemId, $navItemType)
...
+                        $pageModel = new NavItemPage();
+                        $pageModel->attributes = [
+                            'nav_item_id' => $navItemId,
+                            'timestamp_create' => time(),
+                            'create_user_id' => Yii::$app->adminuser->getId(),
+                            'version_alias' => Module::t('Initial'),
+                            'layout_id' => Yii::$app->request->post('layout_id'),
+                        ];
+                        if (!$pageModel->save())
+                        {
```
Comment (nadar) [inline src/admin/apis/NavItemController.php]: psr2 ;-)

File: src/admin/apis/NavItemController.php [hunk 7]
```diff
@@ -324,16 +324,34 @@ public function actionUpdatePageItem($navItemId, $navItemType)
...
+                            return $this->sendMOdelError($pageModel);
```
Comment (nadar) [inline src/admin/apis/NavItemController.php]: Wrong typing MO ...

File: CHANGELOG.md [hunk 8]
```diff
@@ -13,6 +13,7 @@ All notable changes to this project will be documented in this file. This projec
 ### Fixed
 
 + [#25](https://github.com/luyadev/luya-module-cms/issues/25) Hide deleted pages in dashboard object "last page updates".
++ [#17](https://github.com/luyadev/luya-module-cms/issues/17) Create new page version if changing from other page types to "page" and no version is available.
```
Comment (nadar) [inline CHANGELOG.md]: @martinpetrasch one more thing, 1.0.1 is already released, so you should add this to 1.0.2 changelog.

**Pull request merged**